File: 199-binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
   void find(TreeNode*root,vector<int>&result,int level)
   {
       if(!root)
       {
           return;
       }
        if(result.size()<level)
        {
            result.push_back(root->val);
        }
       find(root->right,result,level+1);
       find(root->left,result,level+1);
   }
    vector<int> rightSideView(TreeNode* root) {
        vector<int>result;
        find(root,result,1);
        return result;
    }
};

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer>result=new ArrayList<>();
        Deque<TreeNode>dq=new ArrayDeque<>();
        if(root!=null)
        {
            dq.offer(root);
        }
        while(!dq.isEmpty())
        {
            int size=dq.size();
            result.add(dq.peekLast().val);
            while(size>0)
            {
                TreeNode temp=dq.peekFirst();
                dq.pollFirst();
                if(temp.left!=null)
                {
                    dq.offer(temp.left);
                }
                if(temp.right!=null)
                {
                    dq.offer(temp.right);
                }
                size--;
            }
        }
        return result;
    }
}